cpp/dolfinx/mesh/EntityMap.h

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2025 Jørgen S. Dokken and Joseph P. Dean
+//
+// This file is part of DOLFINx (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#pragma once
+#include "Topology.h"
+#include <concepts>
+#include <dolfinx/common/IndexMap.h>
+#include <iostream>
+#include <ranges>
+#include <span>
+#include <unordered_map>
+#include <vector>
+
+namespace dolfinx::mesh
+{
+/// @brief A map between entities of two meshes
+class EntityMap
+{
+public:
+  /// @brief Constructor of a map between a set of entities belonging to two
+  /// meshes.
+  ///
+  /// @tparam U
+  /// @param topology A mesh topology
+  /// @param sub_topology Topology of another mesh. This must be a
+  /// "sub-topology" of `topology` i.e. every entity in `sub_topology` must also
+  /// exist in `topology`.
+  /// @param sub_to_topo A list of entities in `topology`. `sub_to_topo[i]` is
```

Comment:
Could the name be longer and better?

Reply:
It's a list representing the map from cells in `sub_topology` to entities in `topology`. How about `sub_topology_to_topology`? Or `sub_to_parent`?